### PR TITLE
[SYCL] Fix vec::convert method.

### PR DESCRIPTION
**Pull request by alexeyvoronov**
Fixed non-compiling code.
Added support for rounding modes on the host device.

Signed-off-by: Alexey Voronov <alexey.voronov@intel.com>

### CODE REVIEW

**Comment by hidekisaito** [comment]
Was the old code for device completely broken? If yes, approved. Else, salvage where floor/ceil was working.

File: sycl/include/CL/sycl/types.hpp [hunk 1]
```diff
@@ -226,17 +228,75 @@ template <typename T> struct LShift {
...
+}
+
+// float to int
+template <typename T, typename R, rounding_mode roundingMode>
+detail::enable_if_t<!std::is_same<T, R>::value && is_float_to_int<T, R>::value,
+                    R>
+convertImpl(T Value) {
+#ifndef __SYCL_DEVICE_ONLY__
+  switch (roundingMode) {
+    // Round to nearest even is default rounding mode for floating-point types
```
Comment (bader) [inline sycl/include/CL/sycl/types.hpp]: I suggest moving host implementation from the headers to the library (i.e. to `source` directory).

**Comment by hidekisaito** [comment]
@alexeyvoronov wrote privately:
Before my changes the code of the convert method didn't compile. 

If this is the case, I don't have any more comments. Please address the comment by @bader where applicable.

File: sycl/include/CL/sycl/types.hpp (hunk 1 above)
Comment (alexeyvoronov) [reply sycl/include/CL/sycl/types.hpp]: I would prefer to do this in the next patch with support for rounding models on the device side .
Then I will make the host and the device part separate and the host part will be moved to the source directory.
I believe that will require some changes,in other files to do this.

File: sycl/test/basic_tests/vec_convert.cpp [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+                              std::is_integral<R>::value),
+                        R>;
+
+int main() {
+  test<int, int, 8, rounding_mode::automatic>(
+      sycl::int8{2, 3, 3, -2, -3, -3, 0, 0},
+      sycl::int8{2, 3, 3, -2, -3, -3, 0, 0});
+  test<float, int, 8, rounding_mode::automatic>(
+      sycl::float8{+2.3f, +2.5f, +2.7f, -2.3f, -2.5f, -2.7f, 0.f, 0.f},
+#ifdef ROUNDING_MODES_IMPLEMENTED
```
Comment (bader) [inline sycl/test/basic_tests/vec_convert.cpp]: Please, remove this macro.
Rounding modes must be supported by `convert` method.

Comment (alexeyvoronov) [reply sycl/test/basic_tests/vec_convert.cpp]: Done.

**Pull request merged**